packages/react-composites/src/composites/CallWithChatComposite/adapter/AzureCommunicationCallWithChatAdapter.ts

```diff
@@ -499,6 +500,97 @@ export const createAzureCommunicationCallWithChatAdapter = async ({
   return new AzureCommunicationCallWithChatAdapter(callAdapter, chatAdapter);
 };
 
+/**
+ * A custom React hook to simplify the creation of {@link CallWithChatAdapter}.
+ *
+ * Similar to {@link createAzureCommunicationCallWithChatAdapter}, but takes care of asynchronous
+ * creation of the adapter internally.
+ *
+ * Allows arguments to be undefined so that you can respect the rule-of-hooks and pass in arguments
+ * as they are created. The adapter is only created when all arguments are defined.
+ *
+ * Note that you must memoize the arguments to avoid recreating adapter on each render.
+ * See storybook for typical usage examples.
+ *
+ * @beta
+ */
+export const useAzureCommunicationCallWithChatAdapter = (
+  /**
+   * Arguments to be passed to {@link createAzureCommunicationCallWithChatAdapter}.
+   *
+   * Allows arguments to be undefined so that you can respect the rule-of-hooks and pass in arguments
+   * as they are created. The adapter is only created when all arguments are defined.
+   */
+  args: Partial<AzureCommunicationCallWithChatAdapterArgs>,
+  /**
+   * Optional callback to modify the adapter once it is created.
+   *
+   * If set, must return the modified adapter.
+   */
+  afterCreate?: (adapter: CallWithChatAdapter) => Promise<CallWithChatAdapter>,
+  /**
+   * Optional callback called before the adapter is disposed.
+   *
+   * This is useful for clean up tasks, e.g., leaving any ongoing calls.
+   */
+  beforeDispose?: (adapter: CallWithChatAdapter) => Promise<void>
+): CallWithChatAdapter | undefined => {
+  const { credential, displayName, endpoint, locator, userId } = args;
+  const [adapter, setAdapter] = useState<CallWithChatAdapter | undefined>(undefined);
+  const afterCreateRef = useRef(afterCreate);
+  const beforeDisposeRef = useRef(beforeDispose);
+
+  // These refs are updated on *each* render, so that the latest values
+  // are used in the `useEffect` closures below.
+  // Using a Ref ensures that new values for the callbacks do not trigger the
+  // useEffect blocks, and a new adapter creation / distruction is not triggered.
+  afterCreateRef.current = afterCreate;
+  beforeDisposeRef.current = beforeDispose;
+
+  useEffect(
+    () => {
+      if (!credential || !displayName || !endpoint || !locator || !userId) {
+        return;
+      }
+      (async () => {
+        let newAdapter = await createAzureCommunicationCallWithChatAdapter({
+          credential,
+          displayName,
+          endpoint,
+          locator,
+          userId
+        });
+        if (afterCreateRef.current) {
+          newAdapter = await afterCreateRef.current(newAdapter);
```

Comment:
what is the `await afterCreateRef.current(newAdapter);`
Could this not be awaited and just be
`afterCreateRef.current = newAdapter`;

And what does assigning it back to newAdapter do?

Reply:
never mind I see now!